File: backend/src/main/java/com/codeloom/backend/dao/topic/TopicRepositoryCustom.java
package com.codeloom.backend.dao.topic;

import com.codeloom.backend.model.*;
import java.util.Collection;

public interface TopicRepositoryCustom {
  Iterable<Topic> findByProblemId(long id);

  void saveAllProblemRelationships(Collection<ProblemTopicRelationship> r);

  void deleteRelationshipsWithProblem(long id);
}

File: backend/src/main/kotlin/com/codeloom/backend/dao/topic/TopicRepositoryCustom.kt
package com.codeloom.backend.dao.topic

import com.codeloom.backend.model.ProblemTopicRelationship
import com.codeloom.backend.model.Topic

interface TopicRepositoryCustom {
    fun findByProblemId(problemId: Long): Iterable<Topic>
    fun saveAllProblemRelationships(problemTopicRelationships: Collection<ProblemTopicRelationship>)
    fun deleteRelationshipsWithProblem(problemId: Long)
}